.circleci/config.yml

```diff
@@ -10,27 +10,44 @@ jobs:
       - run:
           name: Check formatting
           command: ./scripts/rustfmt.sh
-          working_directory: .
   build:
     docker:
       - image: maxsam4/rust
     environment:
       VERBOSE: "1"
     steps:
       - checkout
+      - run:
+          name: Build concatenated lock file for caching
+          command: find . -name "Cargo.lock" -not -path "*/target/*" | xargs cat > concatenated.lock
+      - run:
+          name: Store rust version in an environment var for cache key
+          command: rustc --version > rust.version
       - restore_cache:
           keys:
-            - build-cache-{{ checksum "./Cargo.lock" }}
-      - run:
-          name: Build rocksdb
-          command: cargo build -p rocksdb -j 1 --release
-          working_directory: .
+            - release-cache-{{ checksum "./concatenated.lock" }}-{{ checksum "./rust.version" }}
+            - release-cache2-{{ checksum "./Cargo.lock" }}-{{ checksum "./rust.version" }}
+            - release-cache-{{ checksum "./rust.version" }}
       - run:
           name: Build release
           command: cargo build -j 1 --release
-          working_directory: .
+          no_output_timeout: 4h
```

Comment:
Not really, there's no real downside though. The default timeout of 10 minutes is not enough. We need something like 20 minutes right now but it may increase in the future. Setting it to something this high just gives us more leeway.